### ISSUE TITLE
Noise when recording voice from safari browser

### ISSUE THREAD

**Issue by Purnendu220** [collab-project/videojs-record#248]
## Description
I have implemented audio record feature in my website.But audio record result me recorded audio with noise.
## Steps to reproduce
go to https://qa.myu.co on safari browser.
then login using username and password 
use username:gituser
use password:123456

then u will find the audio icon click on that than use it to record audio.

## Results
After that when you replay the audio you recorded than you will find the issue with the recorded audio.
### Expected
Need recorded voice to be clear 
### Actual
The recoded audio on safari have noise so i can't understand the audio.

## Additional Information
Please include any additional information necessary here. Including the following:
### versions
VIDEOJS: Using video.js 6.9.0 with videojs-record 2.1.2 + videojs-wavesurfer 2.2.2 and recordrtc 5.4.6

#### browsers
safari
#### platform
what platforms (operating systems and devices) are affected?
mac

###Code used by me is
```javascript
 var obj = document.createElement('audio');
      $(obj).attr('id', 'myAudio');
      $(obj).attr('class', 'video-js vjs-default-skin');
      $(obj).attr('style', 'width:80%; margin:auto;');
      $("#rec_audio_div").append(obj);
      playerAudio = videojs("myAudio", {
        controls: true,
        width: 400,
        height: 200,
        controlBar: {
          fullscreenToggle: false,
          volumePanel: false
        },
        plugins: {
          wavesurfer: {
            src: "live",
            waveColor: "#36393b",
            progressColor: "black",
            debug: true,
            cursorWidth: 1,
            msDisplayMax: 20,
            hideScrollbar: true
          },
          record: {
            audio: true,
            video: false,
            maxLength: that.audioVideoLimit
          }
        }
      },  function() {
    // print version information at startup
    videojs.log('Using video.js', videojs.VERSION,
        'with videojs-record', videojs.getPluginVersion('record'),
        '+ videojs-wavesurfer', videojs.getPluginVersion('wavesurfer'),
        'and recordrtc', RecordRTC.version);
  });
  playerAudio.on('deviceError', function() {
    that.alertOnDeviceError(playerAudio.deviceErrorCode);
    console.log('device error:', playerAudio.deviceErrorCode);
  });
  playerAudio.on('error', function(error) {
    that.alertOnDeviceError(error);
    console.log('error:', error);
  });
  // user clicked the record button and started recording
  playerAudio.on('startRecord', function() {
    console.log('started recording!');
    that.state = 1;
    progress=0;
    startAudioTime();
  
  });
  // user completed recording and stream is available
  playerAudio.on('finishRecord', function() {
    that.state = 2;
    clearInterval(recordTimer);
  });
```

**Comment by thijstriemstra** [comment]
I think this is a browser issue, and not something I can do anything about, see this similar report: https://stackoverflow.com/questions/49612970/audio-recording-using-safari-desktop

**Comment by Purnendu220** [comment]
thanks for the response this demo is working fine

**Comment by thijstriemstra** [comment]
> I am little bit confused how I can use this in my angular 2 project

Can you reproduce it with https://collab-project.github.io/videojs-record/examples/audio-only.html?

**Comment by dirkk0** [comment]
I remember that this example was broken for me in the past (i.e. crackling noises).
But I can confirm that it now works on an 2018 iPad with iOS 11.3.

**Comment by thijstriemstra** [comment]
@dirkk0 did you try with latest? that could've fixed it because there was a high CPU issue in videojs-wavesurfer <= 2.4.0. Thanks for the feedback!

**Comment by thijstriemstra** [comment]
@Purnendu220 please try the demo again (or use latest videojs-record and videojs-wavesurfer in your project), it should be fixed now.

**Comment by Purnendu220** [comment]
Thanks for the support.

**Comment by dirkk0** [comment]
Just for the records: it doesn't crackle on an 2018 iPad with iOS 11.3, as reported above.

It *does* crackle on Safari Version 11.1.1 on Mac OSX 10.12.6, unfortunately.

**Comment by thijstriemstra** [comment]
@dirkk0 what versions of all libraries?

**Comment by dirkk0** [comment]
Good point.
I am not entirely sure that I tested it with the latest version.
I'll check out the latest version tomorrow and report back.

**Comment by dirkk0** [comment]
I tested it with
"VIDEOJS: – "Using video.js 7.0.3 with videojs-record 2.3.1 + videojs-wavesurfer 2.5.0 and recordrtc 5.4.6""
(all drawn with 'yarn add')
and with the version at this link:
https://collab-project.github.io/videojs-record/examples/audio-only.html
VIDEOJS: Using video.js 7.0.3 with videojs-record 2.3.1 + videojs-wavesurfer 2.5.0 and recordrtc 5.4.7

Crackles on Safari 11/MacOSX, but clear sound on iPad, which is surprising.

**Comment by thijstriemstra** [comment]
Thx for feedback. No idea what causes this (besides it being shit browser).

**Comment by thijstriemstra** [comment]
This looks related: https://bugs.webkit.org/show_bug.cgi?id=154538

**Comment by thijstriemstra** [comment]
@dirkk0 can you try with different audio quality settings and see if that changes things?

**Comment by thijstriemstra** [comment]
Thanks for the feedback @dirkk0. Does this example produce noise on safari? https://www.webrtc-experiment.com/RecordRTC/ Select 'Microphone' and ogg or pcm.

**Comment by dirkk0** [comment]
Actually it doesn't work at all in Safari11/MacOSX.
You can only chose 'Microphone' and only 'ogg' and after recording, nothing happens (i.e. I don't hear anything).
The error on the console is:
"Unhandled Promise Rejection: AbortError: The operation was aborted." RecordRTC: 1550

On Chrome, it works fine.

**Comment by thijstriemstra** [comment]
@Purnendu220 and @dirkk0, can you give this a try? https://github.com/collab-project/videojs-record/issues/295#issuecomment-429085457

**Comment by dirkk0** [comment]
@thijstriemstra yes, it works: Safari 12 Mac OSX, and Safari 11 ( I think, I am still on iOS 11.3.)

**Comment by dirkk0** [comment]
Yes, it works without noise on MacOSX/Safari12.
Interestingly enough, the voice is downsampled on iOS now (i.e. slower and deeper).
